### PR TITLE
Keycard Authenticator Devices have no delay and now provide actual feedback.

### LINKED ISSUE #26298: No ERT Swipe Confirmation
(WEB REPORT BY: west3436 REMOTE: 206.221.180.138:7777)
# Revision
666fecf41c0e0084753378dedac29944bbb7f9e5

# Description
Swiping for an ERT does not give you any confirmation that the swipe went through.

# Steps to Reproduce
Swipe for an ERT.

# What you Expected
Some sort of notice that the reqest has been sent.

# What Actually Happened
Silence

### PR DESCRIPTION
**Pull request by RaccoonBro**
[tweak]

The keycard auth devices in head offices will now provide feedback when you swipe your card, tell you that an ERT was successfully requested, and will now instantly approve the alert/access/ERT when a second person swipes, instead of waiting a few seconds.

Fixes #26298

:cl:
 * tweak: Keycard Authentication Devices now have no delay between the second person swiping and having something happen.
 * tweak: Requesting an ERT with the keycard device will now display a message if you successfully request one.

### CODE REVIEW

**Comment by RaccoonBro** [comment]
the code is probably shit but it's tested working so hey

File: code/modules/security levels/keycard authentication.dm [hunk 1]
```diff
@@ -45,9 +45,22 @@ var/global/list/obj/machinery/keycard_auth/authenticators = list()
...
+					confirmed = 1
+
+					// All the useful information is in the source device, so copy it over. There's probably a better way to do this.
+					event_triggered_by = event_source.event_triggered_by
+					event_confirmed_by = usr
+					event = event_source.event
+					ert_reason = event_source.ert_reason
+
+					playsound(src, get_sfx("card_swipe"), 60, 1, -5)
+					to_chat(user, "You swipe your ID card to confirm the [event].")
```
Comment (ShiftyRail) [inline code/modules/security levels/keycard authentication.dm]: Wrap this in `<span class='notice'>`.

File: code/modules/security levels/keycard authentication.dm [hunk 2]
```diff
@@ -45,9 +45,22 @@ var/global/list/obj/machinery/keycard_auth/authenticators = list()
...
+
+					trigger_event(event)
 			else if(screen == 2)
+				playsound(src, get_sfx("card_swipe"), 60, 1, -5)
+				to_chat(user, "You swipe your ID card to request the [event].")
```
Comment (ShiftyRail) [inline code/modules/security levels/keycard authentication.dm]: Same as above.

Comment (RaccoonBro) [reply code/modules/security levels/keycard authentication.dm]: Should the existing to_chat stuff be notices as well? Telling you if it isn't powered n stuff

Comment (ShiftyRail) [reply code/modules/security levels/keycard authentication.dm]: Probably better.

**Comment by GreatBigFailure** [comment]
I've never been sure if I was successfully swiping or if someone else just swiped instead. I've been pretending to know how this whole time.

**Comment by RaccoonBro** [comment]
Small tweak to let IDs in PDAs be used

File: code/modules/security levels/keycard authentication.dm [hunk 3]
```diff
@@ -186,8 +188,15 @@ var/global/list/obj/machinery/keycard_auth/authenticators = list()
...
 			var/datum/striketeam/ert/response_team = new()
 			response_team.mission = ert_reason
-			response_team.trigger_strike()
+			response_team.trigger_strike() // This will show up as *null* in logs, adding requester gives them power to approve an ERT.
 			feedback_inc("alert_keycard_auth_ert",1)
+			event_source.visible_message("<span class='notice'>An ERT has been requested, please await a response.</span>")
+			playsound(event_source, 'sound/machines/notify.ogg', 60, 0)
+
+	log_game("[key_name(event_triggered_by)] triggered and [key_name(event_confirmed_by)] confirmed event [event][event=="Emergency Response Team"?". ERT reason given was '[ert_reason]'":""]")
+	message_admins("[key_name(event_triggered_by)] triggered and [key_name(event_confirmed_by)] confirmed event [event][event=="Emergency Response Team"?". ERT reason given was '[ert_reason]'":""]", 1)
```
Comment (DamianX) [inline code/modules/security levels/keycard authentication.dm]:
```suggestion
	message_admins("[key_name(event_triggered_by)] triggered and [key_name(event_confirmed_by)] confirmed event [event][event=="Emergency Response Team"?". ERT reason given was '[ert_reason]'":""]")
```
that 1 does nothing

**Pull request merged**